onnxruntime/core/framework/bfc_arena.cc

```diff
@@ -411,6 +433,79 @@ void BFCArena::Free(void* p) {
   }
 }
 
+Status BFCArena::Shrink() {
+  std::lock_guard<OrtMutex> lock(lock_);
+  auto num_regions = region_manager_.regions().size();
+  std::vector<void*> region_ptrs;
+  std::vector<size_t> region_sizes;
+  region_ptrs.reserve(num_regions);
+  region_sizes.reserve(num_regions);
+
+  if (consider_first_allocation_region_for_shrinkage_) {
+    // Consider all regions for shrinkage
+    for (const auto& region : region_manager_.regions()) {
+      region_ptrs.push_back(region.ptr());
+      region_sizes.push_back(region.memory_size());
+    }
+  } else {
+    for (const auto& region : region_manager_.regions()) {
+      // Consider only the non-initial regions for shrinkage
+      if (region.id() != 0) {
+        region_ptrs.push_back(region.ptr());
+        region_sizes.push_back(region.memory_size());
+      }
+    }
+  }
+
```

Comment:
nit:  Could combine these and just make it `if (consider_first_allocation_region_for_shrinkage_ || region.id() != 0

Reply:
done